doc/swagger-api-description/api.yaml

```diff
@@ -173,69 +175,159 @@ paths:
           schema:
             $ref: '#/definitions/ErrorModel'
 
-definitions:
+  /clouds:
+    post:
+      # THE PROBLEM: we need to send an existing User and a new Cloud for him. How should we link togeather User and Cloud in the parameters???
+      description: Allows one to add an external cloud account to an existing ariADDna's user. The User MUST be already registered at the cloud service to be added.
+      operationId: addExternalCloudAccount
+      parameters:
+        - in: body
+          name: cloud
+          schema:
+           $ref: '#/definitions/Cloud'
+      responses:
+        '201':
+          description: New account was added successfully.
+          #It's a bad schema idea... What do we expect to get back?
+          schema:
+            $ref: '#/definitions/Cloud'
 
+    get:
+      description: Returns a list of clouds which a certain user has connected to his ariADDna account.
+      operationId: getExternalCloudsList
+      parameters:
+        - in: body
+          name: user
+          schema:
+            $ref: '#/definitions/User'
+      responses:
+        '200':
+          description: The list was found.
+          #Again here schema is just a stub. I don't know how to represent a list of clouds in the response body.
```

Comment:
Pagination - not return a complete set of objects in response, and a certain number of request.
For example Common Page object:
```YAML
Uri:
    type: object
    properties:
      title:
         type: string
      href:
         type: string
      methods:
         type: array
         items:
           type: string
           enum:
             - GET
             - PUT
             - POST
             - DELETE
UriSet:
    type: array
    items:
       $ref: "#/definitions/Uri"

 Page:
    type: object
    properties:
      offset:
        type: integer
        description: offset of first entry
      limit:
        type: integer
        description: number of entries
      entries:
        $ref: "#/definitions/UriSet"
```
Specific Page object based on Common object but use SpecificObjectSet and SpecificObject instead UriSet and Uri.

```YAML
Credential:
  type: object
  description: Credential for a basic user/password authentication.
  properties:
    user:
      type: string
      description: User's login.
    password:
      type: string
      description: A password associated with the login.
      format: password

CredentialsSet:
    type: array
    items:
       $ref: "#/definitions/Credential"

 CredentialsPage:
    type: object
    properties:
      offset:
        type: integer
        description: offset of first entry
      limit:
        type: integer
        description: number of entries
      entries:
        $ref: "#/definitions/CredentialsSet"
```